apps/remote/src/app/app.tsx

```diff
@@ -1,11 +1,9 @@
-import { DataContext } from '@micro-frontend-react/shared/data-context';
+import { CounterContextType, DataContext } from '@micro-frontend-react/shared/data-context';
 import { useContext } from 'react';
 import { CustomButton } from '@micro-frontend-react/common-ui';
 
-type CounterState = { counter: number; decrement: () => {} };
-
 const App = () => {
-  const { counter, decrement } = useContext(DataContext) as CounterState;
+  const { counter, decrement } = useContext<CounterContextType>(DataContext)
```

Comment:
We should use the keyword `as` only in very edge cases. It's a bit dangerous.

You did not need to create a new type `CounterState`. There was already `CounterContextType`. All that was needed was export it in the library and import it here.